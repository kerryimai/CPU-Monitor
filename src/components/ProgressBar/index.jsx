import React, { Component } from "react";
import { PropTypes as T } from "prop-types";
import classNames from "classnames";
import styles from "./styles.module.css";

class ProgressBar extends Component {
  static propTypes = {
    completion: T.number,
    label: T.string
  };

  getPercentage = () => {
    const val = (this.props.completion / this.props.CPUCounts) * 100;
    return parseFloat(val.toFixed(2));
  };
  render() {
    const percent = this.getPercentage();
    const style = { width: `${percent <= 100 ? percent : 100}%` };

    return (
      <div className={styles.progressbarWrapper}>
        <div className={styles.progressbarLabel}>{this.props.label} min</div>
        <div className={styles.progressbar}>
          <div
            className={classNames({
              [styles.completion]: percent <= 100,
              [styles.over]: percent > 100
            })}
            style={style}
          />
        </div>
        <div className={styles.progressNum}>{percent || 0}%</div>
      </div>
    );
  }
}

export default ProgressBar;
