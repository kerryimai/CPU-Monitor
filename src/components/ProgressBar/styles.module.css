/*Styling for progress circle start here:*/
.progressbar {
  background: #393738;
}

.completion {
  transition: width 1s ease;
  background: linear-gradient(to right, #3abbe7, #aa9daf);
}

.over {
  transition: width 1s ease;
  background: linear-gradient(to right, #ce4141, #aa9daf);
}

.progressbar,
.completion,
.over {
  height: 6px;
  border-radius: 3px;
  z-index: 2;
  width: 65%;
}

.progressbarWrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 96%;
  height: 10%;
}

.progressbarLabel {
  text-align: left;
  font-weight: 100;
  color: #b7b7b7;
  font-size: 14px;
  width: 28%;
}

.progressNum {
  color: #ce9797;
  font-size: 18px;
  padding-left: 12px;
  width: 100px;
  text-align: left;
}

@media (max-width: 770px) {
  .progressbarWrapper {
    justify-content: center;
    align-items: center;
    width: 100%;
  }
}
