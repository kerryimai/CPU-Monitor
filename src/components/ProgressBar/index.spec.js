import React from "react";
import ProgressBar from "./";
import { shallow } from "enzyme";

it("renders", () => {
  const wrapper = shallow(<ProgressBar />);
  expect(wrapper.html()).toMatchSnapshot();
});
