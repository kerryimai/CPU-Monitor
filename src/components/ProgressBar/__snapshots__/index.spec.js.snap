// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `"<div class=\\"progressbarWrapper\\"><div class=\\"progressbarLabel\\"> min</div><div class=\\"progressbar\\"><div class=\\"\\" style=\\"width:100%\\"></div></div><div class=\\"progressNum\\">0%</div></div>"`;
